### PR TITLE
[bashible-apiserver] Specify TLS cipher suites

### LINKED ISSUE #4468: Prevent usage of vulnerable tsl ciphers.
### Preflight Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/deckhouse/deckhouse/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://github.com/deckhouse/deckhouse/issues) for an issue that matches the one I want to file, without success.

### Use case. Why is this important?

By default deckhouse components use the standard tls cipher list (inherited from the go crypto/tls library) which includes the vulnerable 3des. Several security scanners give us an error about insecure algorithms. We want to have ability to change list of used tsl ciphers for deckhouse components.

### Proposed Solution

Remove insecure tls ciphers from go code of deckhouse components.
Phase 1:
- [ ] Remove insecure tls ciphers from [kube-rbac-proxy](https://github.com/deckhouse/deckhouse/tree/main/modules/000-common/images/kube-rbac-proxy).
      * create patch to do not use 3des cipher by default.
      * create patch to avoid use of bash entrypoint (check secure-listen-address has ip is set from code).
      * remove bash entrypoint.
      * bump version of kube-rbac-proxy.

- [ ] Remove insecure tls ciphers from [bashible-apiserver](https://github.com/deckhouse/deckhouse/blob/main/modules/040-node-manager/images/bashible-apiserver).

Phase 2:
- [ ] after merge https://github.com/deckhouse/deckhouse/pull/4635 move kube-rbac-proxy to use common/distroless image as base.
- [ ] after merge https://github.com/deckhouse/deckhouse/pull/4635 move bashible-apiserver to use common/distroless image as base.

Phase 3:
- [ ] Remove insecure tls ciphers from another deckhouse components.

### Additional Information
```
| ssl-enum-ciphers:
|   TLSv1.2:
|     ciphers:
|       TLS_RSA_WITH_AES_128_GCM_SHA256 (rsa 2048) - A
|       TLS_RSA_WITH_AES_256_GCM_SHA384 (rsa 2048) - A
|       TLS_RSA_WITH_AES_128_CBC_SHA (rsa 2048) - A
|       TLS_RSA_WITH_AES_256_CBC_SHA (rsa 2048) - A
|       TLS_RSA_WITH_3DES_EDE_CBC_SHA (rsa 2048) - C
|     compressors:
|       NULL
|     cipher preference: server
|     warnings:
|       64-bit block cipher 3DES vulnerable to SWEET32 attack
|       Forward Secrecy not supported by any cipher
```

Need to disable TLS_RSA_WITH_3DES_EDE_CBC_SHA и TLS_ECDHE_RSA_WITH_3DES_EDE_CBC_SHA.

### PR DESCRIPTION
**Pull request by artbin**
## Description

Specify TLS cipher suites used by `bashible-apiserver`.

## Why do we need it, and what problem does it solve?

Prevent usage of vulnerable TLS ciphers in `bashible-apiserver`.
Close #4468

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: node-manager
type: fix
summary: Prevent usage of vulnerable TLS ciphers in `bashible-apiserver`.
impact_level: default
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @artbin. I've started the workflow for label 'e2e/run/openstack'...

File: modules/040-node-manager/images/bashible-apiserver/pkg/cmd/server/start.go [hunk 1]
```diff
@@ -52,6 +52,22 @@ func NewBashibleServerOptions(out, errOut io.Writer) *BashibleServerOptions {
 		StdErr: errOut,
 	}
 	o.RecommendedOptions.Etcd = nil
+	o.RecommendedOptions.SecureServing.CipherSuites = []string{
```
Comment (RomanenkoDenys) [inline modules/040-node-manager/images/bashible-apiserver/pkg/cmd/server/start.go:55]: Why do yo build static list instead of using CipherSuites() func ?

Comment (artbin) [reply modules/040-node-manager/images/bashible-apiserver/pkg/cmd/server/start.go:55]: If we use `tls.CipherSuites()` then the http2 library returns an error `main.go:23 error configuring http2: http2: TLSConfig.CipherSuites index 5 contains an HTTP/2-approved cipher suite (0x1301), but it comes after unapproved cipher suites. With this configuration, clients that don't support previous, approved cipher suites may be given an unapproved one and reject the connection.`

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @artbin. I've started the workflow for label 'e2e/run/openstack'...

**Review by RomanenkoDenys — APPROVED** [review]
LGTM

**Comment by deckhouse-BOaTswain** [comment]
Cherry pick PR [4867](https://github.com/deckhouse/deckhouse/pull/4867) to the branch [release-1.46](https://github.com/deckhouse/deckhouse/tree/release-1.46) successful!

**Pull request merged**